shell.nix

```diff
@@ -45,10 +57,36 @@ let inherit (nixroot) stdenv pkgs lib
     '';
   };
 
+  libkeyfinder = stdenv.mkDerivation {
+    name = "libkeyfinder";
+    version = "2.2.1-dev";
+
+    src = fetchgit {
+      url = "https://github.com/ibsh/libKeyFinder.git";
+      rev = "9b4440d66789b06483fe5273e06368a381d22707";
+      sha256 = "008xg6v4mpr8hqqkn8r5y5vnigggnqjjcrhv5r6q41xg6cfz0k72";
+    };
+
+    nativeBuildInputs = [
+      cmake
+    ];
+    buildInputs = [
+      fftw
+      pkgconfig
+    ];
+
+    buildPhase = ''
+      make VERBOSE=1 keyfinder
+    '';
+    installPhase = ''
+      make install/fast
+    '';
+  };
+
   shell-configure = nixroot.writeShellScriptBin "configure" ''
     mkdir -p cbuild
     cd cbuild
-    cmake .. "$@"
+    cmake .. ${lib.strings.concatStringsSep " " cFlags} "$@"
```

Comment:
This does not work if an element of `cFlags` contains a space. The correct expression is `lib.escapeShellArgs cFlags`, I think.